samcli/local/docker/container.py

```diff
@@ -272,6 +283,35 @@ def start(self, input_data=None):
         # Start the container
         real_container.start()
 
+        # Wait for port to be open
+        self.wait_for_port()
+
+    def wait_for_port(self):
+        """
+        Waits until the host machine port that Docker binds to is open.
+        """
+        start_time = time.time()
+        sleep = 0.1
+        while True:
+            a_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            location = (self._container_host, self.rapid_port_host)
```

Comment:
changed to `self._container_host` to match the host used on the invoke: https://github.com/aws/aws-sam-cli/blob/develop/samcli/local/docker/container.py#L282